lib/navigation/preprocess_keyence.hpp

```diff
@@ -3,18 +3,23 @@
 #include <array>
 #include <cstdint>
 
+#include "core/logger.hpp"
 #include "core/types.hpp"
 
 namespace hyped::navigation {
 
 class KeyencePreprocessor {
  public:
-  KeyencePreprocessor();
+  /**
+   * @brief Construct a new Keyence Preprocessor object
+   */
+  KeyencePreprocessor(hyped::core::ILogger &log_);
 
   SensorChecks checkKeyenceAgrees(const core::KeyenceData &keyence_data);
 
  private:
-  bool is_keyence_agreeing_;
+  hyped::core::ILogger &log_;
+  bool has_keyence_disagreed_;
```

Comment:
comment describing briefly purpose of this flag

Reply:
`last_keyence_measurement_was_in_disagreement_`? Or is this getting to verbose? I find boolean flags terribly difficult to name.

Here's a better approach: make an enum class, say `KeyenceAgreement`, and call the field `last_measurement_agreement_`? The values of `KeyenceAgreement` could be `kAgree` and `kDisagree` or similar.